Refatorando classe Animal e Banco de Dados

01lucas-costa opened the pull request:
Aplicando recursos  de upload de imagens
Aplicando Procedure e Triggers no Mysql
Será cadastrados somente o nomes da imagens dos animais em uma tabela separada(imagens_animais)

01lucas-costa committed:
Adicionando classe de Upload de imagens

Agora o método para inserir animais tem dois parâmetros um para o nome da imagem, e outro que recebe o array $_POST .

01lucas-costa committed:
Refatorando o banco de dados

01lucas-costa committed:
Aplicando mudanças em função a refatoração do BD

01lucas-costa committed:
Adicionando recursos no MySQL

Procedure: para facilitar a inserção de dados e para fazer consultas relacionadas entre duas tabelas, sem ter que colocar muito código sql no php.
Trigger: para fazer atualização entre tabelas relacionadas, durante uma exclusão de um registro por exemplo.

01lucas-costa committed:
Classe para genrenciar imagens de animais

Essa classe classe realiza inclusão e exclusão de imagens de um animal já cadastro na tabela animais, com seus end-poinst são para post e delete no index.php

01lucas-costa committed:
Refatorando classe Animal

Trocando códigos sql por procdures do MySQL.
E a função trash() para tratar a exclusão de uma ou várias imagens por animal, e, excluindo também as imagens do diretório.

01lucas-costa committed:
Adicionando classe ImagemDenuncia

Classe responsável por fazer a inserção o exclusão de determinda imagem por uma denuncia já existente.
E seus end-points de post e delete no index

01lucas-costa committed:
Implementando upload de imagens para denuncias

Todoas mundas necessárias no código php e no banco, com criação de procedures e triggers

learnjsfatec commented:
Lucão, o que acha de utilizarmos apenas uma tabela para imagens ao invés de duas? 

Para saber se a imagem é do animal ou denúncia, basta criar uma coluna (flag) na própria tabela.

01lucas-costa committed:
Corrigindo problema de acentuação

01lucas-costa committed:
Refatorando tabelas de imagens

todas imagens ficaram em uma unica tabela(Imagens), sendo o tipo de imagem identificado pela coluna flag

01lucas-costa committed:
Refatorando procedures e triggers

01lucas-costa committed:
mudanças nas classes para trabalhar com a tabela Imagens
